test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/InteroperabilityTest.scala

```diff
@@ -523,6 +523,74 @@ class InteroperabilityTest {
     obj.testAny() // should not throw
   }
 
+  @Test def should_access_global_scope_for_object(): Unit = {
+    assumeTrue("Assuming execution in Node.js", executingInNodeJS)
+
+    nodejs_runInThisContext("""
+      const InteroperabilityTestConstObject = {
+        x: 42
+      };
+    """)
+
+    assertEquals("object", js.typeOf(InteroperabilityTestConstObject))
+    assertEquals(42, InteroperabilityTestConstObject.x)
+  }
+
+  @Test def should_access_global_scope_for_class(): Unit = {
+    assumeTrue("Assuming execution in Node.js", executingInNodeJS)
+
+    nodejs_runInThisContext("""
+      class InteroperabilityTestConstClass {
+        constructor(x) {
+          this.x = x;
+        }
+      };
+    """)
+
+    assertEquals("function",
+        js.typeOf(js.constructorOf[InteroperabilityTestConstClass]))
+    val obj = new InteroperabilityTestConstClass(5)
+    assertEquals(5, obj.x)
+  }
+
+  @Test def should_access_global_scope_for_JSGlobalScope_members(): Unit = {
+    assumeTrue("Assuming execution in Node.js", executingInNodeJS)
+
+    nodejs_runInThisContext("""
+      const InteroperabilityTestLetConstGlobals_value = 456;
+      let InteroperabilityTestLetConstGlobals_variable = "hello";
+      const InteroperabilityTestLetConstGlobals_method = (function(x) {
+        return x + 1;
+      });
+    """)
+
+    import InteroperabilityTestLetConstGlobals._
+
+    assertEquals("number", js.typeOf(InteroperabilityTestLetConstGlobals_value))
+    assertEquals(456, InteroperabilityTestLetConstGlobals_value)
+
+    assertEquals("string", js.typeOf(InteroperabilityTestLetConstGlobals_variable))
+    assertEquals("hello", InteroperabilityTestLetConstGlobals_variable)
+    InteroperabilityTestLetConstGlobals_variable = "world"
+    assertEquals("world", InteroperabilityTestLetConstGlobals_variable)
+
+    assertEquals(6, InteroperabilityTestLetConstGlobals_method(5))
+  }
+
+  /** Runs the specified piece of code in the global context.
+   *
+   *  This only works on Node.js. It needs functionality from the `vm` module.
+   *
+   *  This method can be used to declare global let/const/classes. Any other
+   *  attempt to do so (e.g., using a `require`d source file or a `js.eval`)
+   *  would not expose the bindings in the global scope.
+   */
+  private def nodejs_runInThisContext(code: String): Unit = {
+    val vm = js.Dynamic.global.require("vm")
```

Comment:
An interesting side-effect of this new spec is that dynamic references to `require` now actually work :P And also for other magic variables of Node.js such as `module` and `__filename`.